File: src/modules/company/validators/company_existing_validator.py
from src.modules.company.models import Company
from src.shared.exceptions.not_found_exception import NotFoundException


def company_existing_validator(company_id: int):

    company = Company.objects.filter(id=company_id, deleted_at=None).first()

    if company is None:
        raise NotFoundException("Empresa não encontrada!")

File: src/modules/company/validators/company-existing.validator.ts
import { prisma } from '../../../infra/database/prisma.service';
import { NotFoundException } from '../../../shared/exceptions';

export async function companyExistingValidator(companyId: number) {
  const company = await prisma.company.findFirst({
    where: {
      id: companyId,
    },
  });

  if (!company) throw new NotFoundException('Empresa não encontrada!');
}
